src/base/bittorrent/session.cpp

```diff
@@ -2704,11 +2707,19 @@ uint Session::saveResumeDataInterval() const
     return m_saveResumeDataInterval;
 }
 
-void Session::setSaveResumeDataInterval(uint value)
+void Session::setSaveResumeDataInterval(const uint value)
 {
-    if (value != saveResumeDataInterval()) {
-        m_saveResumeDataInterval = value;
+    if (value == m_saveResumeDataInterval)
+        return;
+
+    m_saveResumeDataInterval = value;
+
+    if (value > 0) {
         m_resumeDataTimer->setInterval(value * 60 * 1000);
+        m_resumeDataTimer->start();
```

Comment:
Shouldn't you check if timer is active before start?
Otherwise the timer will restart if it is already started. The pending fastresume-save with the previous duration should be allowed to complete. No?

Reply:
I've thought of it before, if the user monitors the resume files update, he could be confused if qbt still insist on finishing the existing timer interval, restarting timer here could be more consistent from the observable viewpoint IMO.

Reply:
Agree.
Let's see an example. User has 60 minutes interval and changes it to 3 minutes. Should it wait so long to changes take effect?